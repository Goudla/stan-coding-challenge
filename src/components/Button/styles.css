.button {
  color: #fff;
  background-image: linear-gradient(to bottom, #444 0, #333 100%);
  display: inline-block;
  font-weight: 600;
  font-size: 14px;
  padding: 9px 15px;
  text-decoration: none;

  /* HACK styling that is navbar specific in a generic component */
  margin-top: 15px;
  margin-left: 17px;
}

.button:hover,
.button:focus {
  background-image: linear-gradient(to bottom, #555 0, #444 100%);
}
