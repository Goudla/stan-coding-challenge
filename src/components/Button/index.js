import * as React from 'react';
import { Link } from "react-router-dom";

// Styles
import './styles.css';

type Props = {
  children: React.Node
};

class Button extends React.PureComponent<Props> {
  render() {
    const { children } = this.props;

    return (
      <Link to="/" className="button">
        {children}
      </Link>
    );
  }
}

export default Button;
