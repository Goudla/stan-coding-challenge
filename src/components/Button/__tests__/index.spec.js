import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';

// Scenes
import Button from '../';

describe('<Button />', () => {
  it('Should render', () => {
    const renderer = new ShallowRenderer();
    renderer.render(<Button>Test</Button>);
    const result = renderer.getRenderOutput();
    expect(result).toMatchSnapshot();
  });
});
