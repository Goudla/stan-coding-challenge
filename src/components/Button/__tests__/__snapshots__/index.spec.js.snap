// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> Should render 1`] = `
<Link
  className="button"
  to="/"
>
  Test
</Link>
`;
